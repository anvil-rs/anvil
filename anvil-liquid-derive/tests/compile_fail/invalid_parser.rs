// This file tests the error when a template has an invalid parser expression
// Expected error: "Parser expression must be a valid identifier"

use anvil_liquid_derive::Template;
use serde::Serialize;

#[derive(Serialize, Template)]
#[template(path = "test.txt", parser = 123)]  // Using a number instead of an identifier
struct InvalidParserTemplate {
    value: String,
}

fn main() {
    // This will never run, we're only testing the compile-time error
}
