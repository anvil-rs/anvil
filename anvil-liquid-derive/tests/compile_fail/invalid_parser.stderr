error: couldn't read `$DIR/tests/compile_fail/test.txt`: No such file or directory (os error 2)
 --> tests/compile_fail/invalid_parser.rs
  |
  | #[derive(Serialize, Template)]
  |                     ^^^^^^^^
  |
  = note: this error originates in the macro `include_str` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0689]: can't call method `parse` on ambiguous numeric type `{integer}`
 --> tests/compile_fail/invalid_parser.rs
  |
  | #[derive(Serialize, Template)]
  |                     ^^^^^^^^
  |
  = note: this error originates in the derive macro `Template` (in Nightly builds, run with -Z macro-backtrace for more info)
help: you must specify a concrete type for this numeric value, like `i32`
  |
8 | #[template(path = "test.txt", parser = 123_i32)]  // Using a number instead of an identifier
  |                                           ++++
